test-utils/test-runner/src/node.rs

```diff
@@ -353,30 +209,29 @@ impl<T: ChainInfo> Node<T> {
 
 	/// Revert all blocks added since creation of the node.
 	pub fn clean(&self) {
-		// if a db path was specified, revert all blocks we've added
-		if let Some(_) = std::env::var("DB_BASE_PATH").ok() {
+		// if the initial block number wasn't 0, then revert blocks
+		if self.initial_block_number != 0u32.into() {
```

Comment:
why?

Reply:
One of the use cases of the test runner is running it over live chain data, in this case it's `initial_block_number` is non-zero so we want to revert any blocks we've added, so we can re-use the live chain data.

Reply:
I understand that, but why special-casing initial_block_number == 0? what if started from 0, but still want to revert all blocks?

Reply:
you know what you're right about this one too, it's a test environment, we should always revert.